demo/full/scripts/components/GitHubButton.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from "react";
+import { render } from "github-buttons";
+
+const {
+  useEffect,
+  useRef,
+} = React;
+
+const GitHubButton = ({
+  href,
+  ariaLabel,
+  dataColorScheme = "no-preference: dark_high_contrast; light: dark_high_contrast; dark: dark_high_contrast;",
+  dataIcon,
+  dataShowCount,
+  dataSize = "large",
+  dataText,
+  title,
+  children,
+}: {
+  href: string;
+  ariaLabel?: string;
+  dataColorScheme?: string;
+  dataIcon?: string;
+  dataShowCount?: boolean | string;
+  dataSize?: string;
+  dataText?: string;
+  title?: string;
+  children?: React.ReactNode;
+}): JSX.Element => {
+  const spanRef = useRef<HTMLSpanElement>(null);
+  const aRef = useRef<HTMLAnchorElement>(null);
+
+  useEffect(() => {
+    // Apply github-buttons's render on a new span
+    if (spanRef.current && aRef.current) {
+      const newSpan = spanRef.current.appendChild(document.createElement("span"));
+      render(
+        newSpan.appendChild(aRef.current),
+        el => newSpan.parentNode?.replaceChild(el, newSpan),
+      );
```

Comment:
So if I understand correctly, we:
  1. create a new span and add it as a child of the top span
  2. Move the HTMLLinkElement (`<a>` tag) inside that new inner span (and remove it from where it was before (the top span)
  3. Huh... That `render` call looks weird to me here because from the first argument it looks like we're adding the button inside the link, but the callback actually replace the inner span created in `1`.

It looks like there's unnecessary steps here. Why not:
  1. Just returning a top `span`  in the component
  2. using the other form of the `render` callback where you're just communicating its options
  3. Inserting the button directly inside the top span

Reply:
You're right, I based myself on the logic they implemented for their components but there was indeed some unnecessary steps. 

I couldn't get the other way of calling render to work and it was adding a lot of type complexity (with the optional params that can't be of type `undefined` when passed as the `options`  object), so I kept the `<a>` in the component but the `useEffect` now simply replaces it by the modified one.

Reply:
OK why not.

In that case, we could also kill two birds with one stone (very sadistic expression btw!) by using the default `a` as a fallback if `github-buttons` ever fail (for whatever reason).
For example by defining its for now unused "children" prop.

Kind of like before where it was broken enough to not display buttons, but still functional enough to have links and the right text.

Reply:
Right, I moved the text from the children to the dataText to avoid the visible text while the buttons are loading but having a fallback may be better indeed.

Reply:
As it was inducing layout shifts and visual flickering as the text disappears when the component is loaded, we decided to revert this.